test/test_conversion_driver_esm1p5.py

```diff
@@ -142,3 +152,32 @@ def test_convert_esm1p5_output_dir_error():
         esm1p5_convert.convert_esm1p5_output_dir(
             "/test_convert_esm1p5_output_dir_error/fake/path/"
         )
+
+def test_report_results_success():
+    succeeded_files = ["fake_file_1", "fake_file_2", "fake_file_3"]
+    # succeeded argument to report_results typically a list of Path objects
+    succeeded = [Path(p) for p in succeeded_files] 
+    failed = []
+
+    with mock.patch("builtins.print", return_value = None) as mock_print:
+        esm1p5_convert.report_results(succeeded, failed)
+
+    assert mock_print.call_count == len(succeeded)
+
+def test_report_results_failed():
+    succeeded = []
+    failed = [
+        (Path("fake_file_1"), Exception("Error 1")),
+        (Path("fake_file_2"), Exception("Error 2")),
+        (Path("fake_file_3"), Exception("Error 3"))
+    ]
+
+    with mock.patch("warnings.warn", return_value = None) as mock_warning:
```

Comment:
Does the `with pytest.warns(): ...` context manager work in place of the mock here?

Reply:
I'd used the patching here in order to count the number of times it was called, which possibly was unnecessary. I think we can skip this now though that the printing/warning has been moved up into the main block.